common/trace.c

```diff
@@ -150,11 +183,21 @@ static inline void trace_check_tree(void) {}
 
 static void trace_init(void)
 {
+	const char *dev_trace_file;
 	if (active_spans)
 		return;
-	active_spans = calloc(MAX_ACTIVE_SPANS, sizeof(struct span));
+
+	active_spans = notleak(tal_arrz(NULL, struct span, 1));
 
 	current = NULL;
+	dev_trace_file = getenv("DEV_CLN_TRACE_FILE");
+	if (dev_trace_file) {
+		const char *fname = tal_fmt(tmpctx, "%s.%u",
+					    dev_trace_file, (unsigned)getpid());
+		trace_to_file = fopen(fname, "a+");
+		if (!trace_to_file)
+			err(1, "Opening DEV_CLN_TRACE_FILE %s", fname);
```

Comment:
Bike shedding as usual, but should we adopt `CLN_` as a common prefix for our envvars going forward? It's a minor thing, but could simplify things a bit going forward.

Great addition of a trace file by the way, very useful for simple setups.

Reply:
Hey, trace file was your idea!  And I agree with renaming env vars.